### PR TITLE
Fix search/searchindex deprecations

### PR DESCRIPTION
**Pull request by ararslan**
@nalimilan Does this look right?

### CODE REVIEW

File: src/string/StringBenchmarks.jl [hunk 1]
```diff
@@ -19,13 +19,21 @@ SUITE["join"] = @benchmarkable join($str, $str) time_tolerance=0.40
...
+    g["String"] = @benchmarkable search($str, $("x"))
+
+    g = addgroup!(SUITE, "searchindex")
+    g["Char"] = @benchmarkable searchindex($str,  $('x'))
+    g["String"] = @benchmarkable searchindex($str, $("x"))
+else
+    # The searchindex deprecation target makes the updated searchindex group
+    # redundant with the updated search group, so they've been combined here
+    g = addgroup!(SUITE, "findfirst")
+    g["Char"] = @benchmarkable findfirst(equalto($('x')), $str)
```
Comment (nalimilan) [inline src/string/StringBenchmarks.jl]: I would have thought that with the recent Compat PR this block would also work on 0.6, doesn't it?

Comment (ararslan) [reply src/string/StringBenchmarks.jl]: Yeah I suppose it should. Would it be better then to put only the `search` benchmarks behind a version guard and unconditionally define these for `findfirst`?

Comment (nalimilan) [reply src/string/StringBenchmarks.jl]: I'd just drop `search`, since `findfirst` is implemented in terms of `search` in Compat anyway, so it's covered.

Comment (ararslan) [reply src/string/StringBenchmarks.jl]: I'm not sure what you mean. We still want to benchmark `search` on 0.6.

Comment (nalimilan) [reply src/string/StringBenchmarks.jl]: I mean that Compat implements `findfirst(::EqualTo, ...)` by simply calling `search`, so if you benchmark the former you actually benchmark the latter on 0.6. No need to benchmark it separately.

Comment (ararslan) [reply src/string/StringBenchmarks.jl]: Looks like nothing in Compat calls `searchindex`, so that should stay I guess.

Comment (ararslan) [reply src/string/StringBenchmarks.jl]: I think I'd rather just keep this how it is in this PR. Seems more explicit.

**Pull request merged**